### ISSUE TITLE
About training schedule and weight decay steps

### ISSUE THREAD

**Issue by XCRobert** [GothicAi/Instaboost#18]
You did not adjust the weight decay steps while using 3x and 4x in Detectron.pytorch?
![image](https://user-images.githubusercontent.com/16361120/71575358-9dd62f00-2b27-11ea-94d4-efbedc8af8f7.png)
Fig
![image](https://user-images.githubusercontent.com/16361120/71575415-d5dd7200-2b27-11ea-9735-bea35dda2d58.png)
In Detectron , 1x schedule decay weights after epochs[8,11]. Which steps do you set to decay weights in Fig 7's baseline and in your InstaBoost while using 3x and 4x.

**Comment by Fang-Haoshu** [comment]
Hi， I remember that it was adjusted. You can checkout our configs. https://github.com/GothicAi/Instaboost/tree/master/detectron/InstaBoost_configs
Sorry for the puzzle here.

**Comment by XCRobert** [comment]
Hello! Only InstaBoost_configs can be found. Where are the w/o InstaBoost configs for 4x schedule?

**Comment by Fang-Haoshu** [comment]
Oh I see. The weight decay config w/o InstaBoost should be the same as the 4x schedule.

**Comment by XCRobert** [comment]
Yes, theoretically it should be . It is not unfair if you set decay  steps as follow :
Methods              |        Decay  steps      |   Max_Iteration
w/o InstaBost       |        [0, 60000, 80000]      |            360000
w    InstaBost       |        [0, 240000, 320000]   |            360000

**Comment by Fang-Haoshu** [comment]
Yeah, thus in our exp it's
Methods | Decay steps | Max_Iteration
w/o InstaBost |  [0, 240000, 320000] | 360000
w InstaBost | [0, 240000, 320000] | 360000

Sorry for not making it clear.

**Comment by XCRobert** [comment]
Then , another question : 
To get Figure 7's 16 points in your paper, you train 16 times or 4 times in total ? 
You set decay  steps [0, 60000, 80000], [0, 120000, 160000], [0, 180000, 240000], [0, 240000, 320000]  respectively to get each point in the figure? It will take month-level training time off the reel using 8 Tesla P100 GPUS .
Or , 
you just uniformly set decay steps [0, 240000, 320000] for 12,  24 , 36, 48 epochs?

**Comment by Fang-Haoshu** [comment]
1. It's 8 times. There are no 16 times because detection and segmentation are the same model.
2. We choose the former case you mention. It indeed took a lot of time. In fact this project took nearly a year, or maybe even longer than a year...

**Comment by XCRobert** [comment]
Thanks for your reply. You should alter the claim ' only modifying the training epochs ' in section 4.3 from your paper

**Comment by Fang-Haoshu** [comment]
Yeah, good idea! We will alter that in our journal version paper. Thanks!

**Comment by XCRobert** [comment]
With:  FPN,  Cascade Mask RCNN , ResNeXt 152， multiscale training , DCNv2,  multiscale testing
Your InstaBoost may refresh the COCO instance-segmentation leaderboard

**Comment by Fang-Haoshu** [comment]
Haha, I guess it's something that we can try. It would be encouraging if we can refresh that :D
